libraries/Microsoft.Bot.Connector/Authentication/CertificateServiceClientCredentialsFactory.cs

```diff
@@ -78,7 +80,22 @@ public override Task<bool> IsAuthenticationDisabledAsync(CancellationToken cance
                 throw new InvalidOperationException("Invalid Managed ID.");
             }
 
-            return Task.FromResult<ServiceClientCredentials>(_certificateAppCredentials);
+            // Instance must be reused per audience, otherwise it will cause throttling on AAD.
+            _certificateAppCredentialsByAudience.TryGetValue(audience, out var certificateAppCredentials);
+            if (certificateAppCredentials == null)
+            {
+                certificateAppCredentials = new CertificateAppCredentials(
+                    _certificate,
+                    _appId,
+                    _tenantId,
+                    audience,
+                    _sendX5c,
+                    _httpClient,
+                    _logger);
+                _certificateAppCredentialsByAudience.TryAdd(audience, certificateAppCredentials);
+            }
```

Comment:
Potential concurrency issue. Use [GetOrAdd](https://learn.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2.getoradd?view=netstandard-2.1#examples) to ensure concurrent dictionary thread safety.
```csharp
var certificateAppCredentials = _certificateAppCredentialsByAudience.GetOrAdd(audience, (aud) =>
    new CertificateAppCredentials(
        _certificate,
        _appId,
        _tenantId,
        audience,
        _sendX5c,
        _httpClient,
        _logger)
);
```